service/src/main/java/bio/terra/cbas/initialization/CloneRecoveryService.java

```diff
@@ -39,58 +41,77 @@ public void cloneRecovery() {
         cbasContextConfig.getWorkspaceId(),
         cbasContextConfig.getWorkspaceCreatedDate());
 
-    pruneCloneSourceWorkspaceHistory();
-  }
+    List<Method> clonedMethods =
+        methodDao.getMethods().stream().filter(this::isMethodCloned).toList();
 
-  @WriteTransaction
-  public void pruneCloneSourceWorkspaceHistory() {
-    List<Method> methods = methodDao.getMethods().stream().filter(this::isMethodCloned).toList();
+    List<MethodTemplateUpdateManifest> templateUpdateManifests =
+        clonedMethods.stream()
+            .map(m -> runSetDao.getRunSetsWithMethodId(m.methodId()))
+            .map(this::generateTemplateUpdateManifest)
+            .toList();
 
-    methods.forEach(this::updateMethodTemplate);
+    templateUpdateManifests.stream()
+        .map(MethodTemplateUpdateManifest::templateRunSets)
+        .forEach(rsList -> rsList.forEach(rs -> runSetDao.updateIsTemplate(rs.runSetId(), true)));
 
-    methods.forEach(this::cleanupClonedMethodHistory);
-  }
+    templateUpdateManifests.stream()
+        .map(MethodTemplateUpdateManifest::nonTemplateRunSets)
+        .forEach(rsList -> rsList.forEach(rs -> runSetDao.updateIsTemplate(rs.runSetId(), false)));
 
-  public Boolean isMethodCloned(Method m) {
-    return !m.originalWorkspaceId().equals(cbasContextConfig.getWorkspaceId());
+    clonedMethods.forEach(
+        method -> {
+          List<RunSet> methodRunSets = runSetDao.getRunSetsWithMethodId(method.methodId());
+          getRunSetsToDelete(methodRunSets).forEach(rs -> runSetDao.deleteRunSet(rs.runSetId()));
```

Comment:
Is there a reason for this two-stage "mark it as non-template", then delete it later on?

Could we just delete them straight away on line 59?

Reply:
yes, good thinking